docs/developer-docs/multi-chain/ethereum/evm-rpc/how-it-works.mdx

```diff
@@ -0,0 +1,155 @@
+  Custom : record {
+    chainId : nat64;
+    services : vec record { url : text; headers : opt vec HttpHeader };
+  };
+};
+
+type RpcConfig = record {
+  responseSizeEstimate : opt nat64,
+  ...
+};
+
+type MultiRpcResult<T> = variant {
+  Consistent : Result<T, RpcError>;
+  Inconsistent : vec (
+    variant {  Ethereum : EthereumService;  Sepolia : SepoliaService; ... },
+    Result<T, RpcError>
+  );
+};
+
+eth_getLogs : (RpcServices, RpcConfig, EthGetLogsParam)
+  -> (MultiRpcResult<vec LogEntry>);
+
+eth_getBlockByNumber : (RpcServices, RpcConfig, BlockTag)
+  -> (MultiRpcResult<Block>);
+
+eth_getTransactionReceipt : (RpcServices, RpcConfig, Hash)
+  -> (MultiRpcResult<opt TransactionReceipt>);
+
+eth_getTransactionCount : (RpcServices, RpcConfig, GetTransactionCountArgs)
+  -> (MultiRpcResult<TransactionCount>);
+
+eth_feeHistory : (RpcServices, RpcConfig, FeeHistoryArgs)
+  -> (MultiRpcResult<FeeHistory>);
+
+eth_sendRawTransaction : (RpcServices, RpcConfig, text)
+  -> (MultiRpcResult<SendRawTransactionResult>);
+
+```
+
+The shared Candid types are defined as follows:
+
+- `EthMainnetService`: An RPC service compatible with the Ethereum mainnet.
+
+- `EthSepoliaService`: An RPC service compatible with the Sepolia testnet.
+
+- `RpcServices`: An input for Candid-RPC methods representing which chain and service(s) to use for the RPC call. By default, the canister uses at least 3 different RPC services.
+
+- `RpcConfig`: An optional input for Candid-RPC methods used to customize how the RPC request is performed by the canister.
+
+## General JSON-RPC requests
+
+A general-purpose JSON-RPC endpoint is available for use cases requiring functionality beyond the supported Candid-RPC interface, making it possible to call a wider range of RPC services and EVM blockchains.
+
+This endpoint also offers an improved developer experience when using the [ethers-providers](https://docs.rs/ethers-providers/latest/ethers_providers/) Rust crate or [ethers](https://www.npmjs.com/package/ethers) npm package, which implement strongly-typed functions with convenient type conversions for each JSON-RPC method.
+
+`request`
+
+```
+type RpcService = variant {
+  EthMainnet : EthMainnetService;
+  EthSepolia : EthSepoliaService;
+  Chain : nat64;
+  Provider : nat64;
+  Custom : record { url : text; headers : opt vec HttpHeader };
+};
+
+request : (
+  service : RpcService,
+  jsonRequest : text,
+  maxResponseBytes : nat64
+) -> (
+  Result<text, RpcError>
+);
+```
+
+- `EthMainnet`: Selects a built-in provider for the Ethereum mainnet.
+
+- `EthSepolia`: Selects a built-in provider for the Sepolia testnet.
+
+- `Chain`: Selects a provider from the list of built-in providers with the given chain ID. An extensive list of chain IDs can be found on [ChainList.org](https://chainlist.org).
+
+- `Provider`: Selects the RPC provider with the given id.
+
+- `Custom`: Uses the provided JSON-RPC API information. This option can be used to pass a custom API key.
```

Comment:
maybe worth mentioning that this can also be used to communicate with a local RPC while developing, see [here](https://github.com/letmejustputthishere/chainfusion-starter/blob/c7e73b8fb47d22260c455ae9fc37041a194609d4/deploy.sh#L61).

i was wondering what the header is needed for in `Custom` and what happens if i pick a `Chain` that has no registered provider (i assume i'll receive an error, but wanted to make sure)

Reply:
> i was wondering what the header is needed for in Custom

`header` in `Custom` makes it possible to send HTTP headers with the request, usually for passing the API key. 

> what happens if i pick a Chain that has no registered provider (i assume i'll receive an error, but wanted to make sure)

That's correct. Specifying `Chain` with no registered provider results in an error

Reply:
Added this info